### PR TITLE
Edit form contact details

### PR DESCRIPTION
**Pull request by davidjamesstone**
image

image

image

image

### CODE REVIEW

File: designer/server/src/views/forms/privacy-notice.njk [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{% extends 'layouts/page.njk' %}
+{% extends "layouts/page.njk" %}
```
Comment (colinrotherham) [inline designer/server/src/views/forms/privacy-notice.njk:1]: Thanks for fixing all these single → double quotes

File: designer/server/src/routes/forms/contact/phone.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import * as scopes from '~/src/common/constants/scopes.js'
+import { sessionNames } from '~/src/common/constants/session-names.js'
+import { buildErrorDetails } from '~/src/common/helpers/build-error-details.js'
+import * as forms from '~/src/lib/forms.js'
+import { phoneViewModel } from '~/src/models/forms/contact.js'
+
+export const ROUTE_PATH_EDIT_PHONE_CONTACT =
+  '/library/{slug}/edit/contact/phone'
+
+export const phoneContactSchema = Joi.object().keys({
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply designer/server/src/routes/forms/contact/phone.js:14]: We can't do `Joi.object<FormMetadataContactEmail>()` here. Is there a `.js/jsdoc` equivalent to generics we can use?

File: designer/server/src/views/forms/overview.njk [hunk 3]
```diff
@@ -22,84 +22,79 @@
...
-        }
-      },
-      {
-        key: {
-          text: "Lead organisation"
+          text: "Phone"
         },
         value: {
-          html: form.organisation
+          html: '<div>' + form.contact.phone | escape | nl2br + '</div><br><a href="https://www.gov.uk/call-charges" class="govuk-link govuk-link--no-visited-state" target="_blank">Find out about call charges</a>'
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply designer/server/src/views/forms/overview.njk]: Resolved in 55e35681c49a65ba4359cce1fdc45320e2b57be4

File: designer/server/src/routes/forms/contact/phone.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (colinrotherham) [reply designer/server/src/routes/forms/contact/phone.js:14]: Ah of course

Yeah we do similar things in **forms-manager**

```js
const coll = /** @satisfies {Collection<{draft: FormDefinition}>} */ (
  db.collection(DEFINITION_COLLECTION_NAME)
)
```

But often you have to split them across multiple assignments like in **forms-designer**

```js
const requestByType = /** @type {typeof request<BodyType>} */ (request)
return requestByType('get', url, options)
```

Don't worry for now, but direct support appears to be on the TypeScript 5.7 roadmap:

* https://github.com/microsoft/TypeScript/issues/27387

File: designer/server/src/routes/forms/contact/email.test.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+      auth
+    }
+
+    const { document } = await renderResponse(server, options)
+
+    const address = /** @satisfies {HTMLInputElement | null} */ (
+      document.querySelector('#address')
+    )
+
+    expect(address?.value).toBe('support@defra.gov.uk')
```
[the remark this answers is not in the document]
Comment (davidjamesstone) [reply designer/server/src/routes/forms/contact/email.test.js]: Resolved with 50f25b41dbb00d09c282562b6504b610849d2426

File: designer/server/src/routes/forms/contact/email.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+   */
+  ({
+    method: 'GET',
+    path: ROUTE_PATH_EDIT_EMAIL_CONTACT,
+    async handler(request, h) {
+      const { auth, params, yar } = request
+      const { slug } = params
+      const { token } = auth.credentials
+
+      const validation = yar.flash(sessionNames.validationFailure).at(0)
```
[the remark this answers is not in the document]
Comment (alexluckett) [reply designer/server/src/routes/forms/contact/email.js:37]: This is a really good point. Future improvement?

**Review by alexluckett — APPROVED** [review]

File: designer/server/src/routes/forms/contact/email.test.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (colinrotherham) [reply designer/server/src/routes/forms/contact/email.test.js]: Nice one @davidjamesstone

What about the [`ByRole` queries](https://testing-library.com/docs/queries/byrole/)?

We're currently using field IDs but "role" queries also test if labels/hints etc are wired up too

**Comment by colinrotherham** [comment]
Fixed in [commit range](https://github.com/DEFRA/forms-designer/pull/458/files/737007b4545728b0697c4a66b4b9921641252c90..479e2824f32799283507af2d9ab5ca99fa3d40e8)

**Pull request merged**